### PR TITLE
Add migrations troubleshoot warning

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Some users expect that `--force-reapply` or `--overwrite` will revert the partial result of incorrect migration, yet it is impossible. This patch adds a warning regarding this expectation.

### CODE REVIEW

File: doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst [hunk 1]
```diff
@@ -47,6 +47,13 @@ from etcd using ``tt migrations remove``:
 Incorrect migration applied
 ---------------------------
 
+.. warning::
+
+    Any schema change that was made by incorrect migration before the fail or
+    cancellation must be resolved manually on each replicaset before reapply.
```
Comment (p7nov) [inline doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst]:
```suggestion
    Any schema change that was made by an incorrect migration before its fail or
    cancellation must be resolved manually on each replica set before reapply.
```

File: doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst [hunk 2]
```diff
@@ -47,6 +47,13 @@ from etcd using ``tt migrations remove``:
...
+    Apply flags like ``--force-reapply`` affect only internal migration status and
+    unable to revert migration scenario itself.
```
Comment (p7nov) [inline doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst]:
```suggestion
    ``--force-reapply`` and other ``tt migrations`` options affect only internal status of the migration
    and don't revert changes that it has made in the cluster.
```

**Review by p7nov — COMMENTED** [review]
Please check my suggestions

File: doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst (hunk 1 above)
Comment (DifferentialOrange) [reply doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst]: Updated

File: doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst (hunk 2 above)
Comment (DifferentialOrange) [reply doc/platform/ddl_dml/migrations/troubleshoot_migrations_tt.rst]: Updated

**Review by p7nov — APPROVED** [review]
Great, thanks for the addition.

**Pull request merged**